docs/adr/006_cups.md

```diff
@@ -0,0 +1,285 @@
+
+This is the easiest option for a new user of the Starter Kit, even though it's not the safest one.
+
+When deploying the Starter Kit, an Azure Storage Account is provisioned for Azure Functions.
+
+LoRa Device Provisioning tool should be capable of uploading the CLI provided bundle files and properly update the twin for the concentrator device being created in IoT Hub.
+
+When using this option, the 'Facade' Azure Function will retrieve certificate files from such storage account.
+
+#### b. Azure Key Vault secret
+
+This option requires more manual intervention, as the LoRa Device Provisioning tool is not adapted for uploading the credential bundle files to an Azure Key Vault instance.
+
+Following the documentation on [how to set a secret in Azure Key Vault](https://docs.microsoft.com/en-us/azure/key-vault/secrets/quick-create-portal), you can just create two secrets and retrieve their URLs.
+
+When setting a Key Vault secret in the concentrator twin "cups" section, the 'Facade' Azure Function will retrieve it using a Managed Identity.
+
+#### c. External HTTPS repository
+
+This option requires most manual intervention.
+
+LoRa Device Provisioning tool is not uploading the credential bundle files to the repository.
+
+'Facade' Azure Function can handle a HTTPS endpoint from where the bundles will be downloaded, even though there is no security check or additional AuthN/AuthZ option provided here.
+
+## CUPS Protocol Implementation
+
+### Flow diagram for credential management scenarios
+
+#### Prerequisites
+
+- LoRa Basics™ Station is configured with a cups.uri, a cups.trust and mandatory cups.cert/key
+- The concentrator device twin should include changes as described in following section
+
+#### Flow
+
+```mermaid
+  sequenceDiagram
+    autonumber
+    Concentrator->>CUPS Server: POST /update-info
+    CUPS Server->>IoT Hub: Retrieve 'Concentrator' twin
+    CUPS Server->>CUPS Server: Verifies 'Concentrator' client certificate
+    Note right of CUPS Server: Logic for client certificate validation to be defined. First version including a 'cupsClientThumbprint' to verify against.
+    alt client certificate not valid
+        CUPS Server->>Concentrator: Return forbidden
+    else client certificate valid
+        alt no cups config available
+            CUPS Server->>Concentrator: Return no updates
+        else cups config available
+            alt different cupsUri
+                alt different cupsCredCrc
+                    CUPS Server->>Facade Function: Retrieve CUPS Credential Blob
+                    Facade Function->>CUPS Server: CUPS Credential Blob
+                    CUPS Server->>Concentrator: Updated cupsUri and cupsCred
+                else equal cupsCredCrc
+                    CUPS Server->>Concentrator: Updated cupsUri
+                end
+            else equal cupsUri
+                alt different cupsCredCrc
+                    CUPS Server->>Facade Function: Retrieve CUPS Credential Blob
```

Comment:
Can we make the difference clearer why we have this twice in the flow diagram?

Reply:
Unfortunately github makes your comment hard to understand 😢
What is 'this'? Retrieving CUPS Credential Blob?

Reply:
Yes. It would be helpful to clarify why would need to query it twice (`Retrieve CUPS Credential Blob` is in there twice)